styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/FlyteRunnerHandler.java

```diff
@@ -65,66 +65,85 @@ public FlyteRunnerHandler(FlyteRunner flyteRunner) {
 
   @Override
   public void safeTransitionInto(final RunState state, final EventRouter eventRouter) {
+    if (!flyteRunner.isEnabled()) {
+      var workflowInstance = state.workflowInstance();
+      LOG.error("Unable to transition {} into {}. Flyte system is not available", workflowInstance, state.state());
+      eventRouter.receiveIgnoreClosed(Event.halt(workflowInstance), state.counter());
+      return;
+    }
+
     switch (state.state()) {
       case SUBMITTING:
-        LOG.info("Entered state SUBMITTING for: " + state.workflowInstance());
-
-        final FlyteExecConf flyteExecConf = state.data().executionDescription().orElseThrow().flyteExecConf().orElseThrow();
-        final String executionId = state.data().executionId().orElseThrow();
-        final String execName = styxExecIdToFlyteNameMapper.apply(executionId);
-
-        try {
-          LOG.info("running:{}, conf:{}, state:{}", state.workflowInstance(), flyteExecConf, state);
-          flyteRunner.createExecution(execName, flyteExecConf);
-        } catch (Exception e) {
-          try {
-            final var errMessage = "Failed to start execution for " + state.workflowInstance();
-            LOG.error(errMessage, e);
-            eventRouter.receive(Event.runError(state.workflowInstance(), e.getMessage()), state.counter());
-          } catch (IsClosedException isClosedException) {
-            LOG.warn("Failed to send 'runError' event", isClosedException);
-          }
-          return;
-        }
-
-        // Emit `submitted` _after_ starting execution to ensure that we retry in case of failure.
-        final Event submitted = Event.submitted(state.workflowInstance(), execName,
-            STATIC_RUNNER_ID);
-        try {
-          LOG.info("Issue 'submitted' event for: " + state.workflowInstance());
-          eventRouter.receive(submitted, state.counter());
-        } catch (IsClosedException isClosedException) {
-          LOG.warn("Could not emit 'submitted' event for: " + state.workflowInstance(),
-              isClosedException);
-        }
+        transitionSubmitting(state, eventRouter);
```

Comment:
these refactorings make it a bit more difficult to work in parallel on these tasks, maybe we can try to apply them separately next time